drivers/usb/udc/udc_bflb_bl61x.c

```diff
@@ -0,0 +1,1480 @@
+{
+	struct net_buf *buf;
+	uint32_t remain = 0;
+
+	buf = udc_buf_get(ep_cfg);
+	LOG_DBG("Event end for 0x%02x got buf %lx", ep_cfg->addr, (uintptr_t)buf);
+	if (buf == NULL) {
+		return -ENODATA;
+		return 0;
+	}
+	if (USB_EP_DIR_IS_OUT(ep_cfg->addr)) {
+		remain = udc_bflb_bl61x_ep_remain(dev, udc_bflb_bl61x_ep_get_fifo(ep_cfg));
+		net_buf_add(buf, ep_cfg->mps - remain);
+	} else {
+		net_buf_pull(buf, buf->len);
+	}
+
+	return udc_submit_ep_event(dev, buf, 0);
+}
+
+static void udc_bflb_bl61x_work_handler(struct k_work *item)
+{
+	struct udc_bflb_bl61x_ev *ev;
+	struct udc_bflb_bl61x_data *priv;
+	struct udc_ep_config *ep_cfg;
+	int err;
+
+	priv = CONTAINER_OF(item, struct udc_bflb_bl61x_data, work);
+
+	while ((ev = k_fifo_get(&priv->fifo, K_NO_WAIT)) != NULL) {
+		err = 0;
+		LOG_DBG("dev %p, ep 0x%02x, event %u",
+			ev->dev, ev->ep_addr, ev->event);
+
+		ep_cfg = udc_get_ep_cfg(ev->dev, ev->ep_addr);
+
+		if (unlikely(ep_cfg == NULL)) {
+			err = -ENODATA;
+			LOG_ERR("Unexpected Invalid Endpoint Configuration in Work Queue");
+		} else {
+			switch (ev->event) {
+			case UBFBL61X_EVT_CTRL_END:
+				err = udc_bflb_bl61x_ctrl_evt_end(ev->dev);
+				break;
+			case UBFBL61X_EVT_END:
+				err = udc_bflb_bl61x_ep_evt_end(ev->dev, ep_cfg);
+				udc_ep_set_busy(ep_cfg, false);
+				break;
+			case UBFBL61X_EVT_XFER:
+			default:
+				break;
+			}
+		}
+
+		if (unlikely(err)) {
+			udc_submit_event(ev->dev, UDC_EVT_ERROR, err);
+		}
+
+		k_mem_slab_free(&udc_bflb_bl61x_ev_slab, (void *)ev);
+	}
+}
+
+static int udc_bflb_bl61x_ep_enqueue(const struct device *dev,
+				   struct udc_ep_config *const config,
+				   struct net_buf *buf)
+{
+	struct udc_bflb_bl61x_data *priv = udc_get_private(dev);
+	uint8_t ep_idx = USB_EP_GET_IDX(config->addr);
+
+	LOG_DBG("%p enqueue %p for ep 0x%02x", dev, buf, config->addr);
+
+	if (config->stat.halted) {
+		LOG_DBG("ep 0x%02x halted", config->addr);
+		return 0;
+	}
+
+	if (USB_EP_DIR_IS_OUT(config->addr)) {
+		if (ep_idx == 0) {
+			udc_buf_put(config, buf);
+			udc_bflb_bl61x_ctrl_dout_start(dev, udc_data_stage_length(buf));
+		} else {
+			udc_buf_put(config, buf);
+			priv->ep_in[ep_idx] = false;
+			udc_ep_set_busy(config, true);
+			udc_bflb_bl61x_ep_dout_start(dev, config);
+		}
+	} else if (buf->len == 0) {
+		if (ep_idx == 0) {
+			udc_bflb_bl61x_ctrl_ack(dev);
+			udc_ctrl_submit_status(dev, buf);
+			udc_ctrl_update_stage(dev, buf);
+		} else {
+			udc_bflb_bl61x_ep_ack(dev, ep_idx);
+		}
+	} else {
+		if (ep_idx == 0) {
+			udc_buf_put(config, buf);
+			udc_bflb_bl61x_ctrl_din_start(dev);
+		} else {
+			udc_buf_put(config, buf);
+			priv->ep_in[ep_idx] = true;
+			udc_ep_set_busy(config, true);
+			udc_bflb_bl61x_ep_din_start(dev, config);
+		}
+	}
+	return 0;
+}
+
+static int udc_bflb_bl61x_ep_dequeue(const struct device *dev,
+				   struct udc_ep_config *const cfg)
+{
+	unsigned int lock_key;
+	struct net_buf *buf;
+
+	lock_key = irq_lock();
+
+	buf = udc_buf_get_all(cfg);
+	if (buf) {
```

Comment:
I think the USB area started to anticipate a possible MISRA compliance, which wants `if (expr)` have `expr` be a bool (seen [here](https://docs.zephyrproject.org/latest/contribute/coding_guidelines/index.html), row 85).

```suggestion
	if (buf != NULL) {
```

Reply:
done